app/move/controllers/timeline.js

```diff
@@ -0,0 +1,33 @@
+const populateResources = require('../../../common/lib/populate-resources')
+const presenters = require('../../../common/presenters')
+const moveService = require('../../../common/services/move')
+
+const getTabsUrls = require('./view/view.tabs.urls')
+
+module.exports = async function view(req, res) {
+  const { move } = req
+  const { profile } = move
+
+  const urls = {
+    tabs: getTabsUrls(move),
+  }
+
+  const { assessment_answers: assessmentAnswers = [] } = profile || {}
+
+  const { timeline_events: moveEvents } = await moveService.getById(move.id, {
```

Comment:
This will introduce the frontend calling the API for the move twice with this change.

Can we instead, make a change to [`setMove`](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/903978823a955b1af54c08d3b7e6c07ca4b24820/app/move/middleware.js#L5-L16) to dynamically ask for the relevant includes.

I'd suggest making that middleware do this:

```javascript
  setMove: async (req, res, next, moveId) => {
    if (!moveId) {
      return next()
    }

    try {
      req.move = await moveService.getById(moveId, {
        include: req.moveInclude,
      })
      next()
    } catch (error) {
      next(error)
    }
  },
```

Then a middleware called only for this route which sets that extra include:

```javascript
function setTimelineInclude (req, res, next) {
  req. moveInclude = 'timeline_events'
  next()
}

moveRouter.get('/timeline', protectRoute('move:view'), setTimelineInclude, timeline)
```

I'd also suggest keeping the `populateResources: true` within the service as the controller/middleware shouldn't have to care about that. We know this is just a patch for **only** events until the API supports the correct includes so I'd rather see that isolated to the service.

Reply:
If it's moved in to the service then we have to check every move for unpopulated resources which is totally unnecessary.

And as an isolated pair of commits, it can be reverted in one go.

As for getting round the setMove conundrum, I think this is much cleaner
```js
router.get(
  `/:moveId(${uuidRegex})/timeline`,
  protectRoute('move:view'),
  timeline
)
```

and then in timeline
```js
  const move = await moveService.getById(req.params.moveId, {
    include: 'timeline_events',
    populateResources: true,
  })
```

Reply:
(I already pushed a commit doing so)